### PR TITLE
Refactor fixture `recover_acl_rule` and and use it in module `cacl/test_cacl_function.py`.

### PR DESCRIPTION
**Pull request by yutongzhang-microsoft**
### Description of PR

In module `cacl/test_cacl_function.py`, it will use command `acl-loader delete` to delete all acl rules, no matter if the testbed has DATAACL rules before test or not. If this testbed has DATAACL rules before, this deletion will cause inconsistent between previous running config and current running config, and will cause unnecessary config reload. In PR #9199 , we have a fixture `recover_acl_rule` to recovery acl rules. So in this PR, I refactor this fixture and use this fixture to recover acl rules in module `cacl/test_cacl_function.py`.

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [ ] 202205

#### What is the motivation for this PR?
In module `cacl/test_cacl_function.py`, it will use command `acl-loader delete` to delete all acl rules, no matter if the testbed has DATAACL rules before test or not. If this testbed has DATAACL rules before, this deletion will cause inconsistent between previous running config and current running config, and will cause unnecessary config reload. In PR #9199 , we have a fixture `recover_acl_rule` to recovery acl rules. So in this PR, I refactor this fixture and use this fixture to recover acl rules in module `cacl/test_cacl_function.py`.

#### How did you do it?
Move fixture `recover_acl_rule` to tests/conftest.py and use this fixture to recover acl rules in module `cacl/test_cacl_function.py`.

#### How did you verify/test it?
```
07:34:42 conftest.core_dump_and_config_check      L1893 INFO   | Core dump and config check passed for cacl/test_cacl_function.py
```

### CODE REVIEW

File: tests/common/templates/default_acl_rules.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+                                    "config": {
+                                        "ethertype": "2048",
+                                        "vlan_id": "1000"
+                                    }
+                                },
+                                "input_interface": {
+                                    "interface_ref":
+                                    {
+                                        "config": {
+                                            "interface": "Ethernet12,Ethernet16,Ethernet20,Ethernet24,Ethernet28,Ethernet32,Ethernet36,Ethernet4,Ethernet40,Ethernet44,Ethernet48,Ethernet52,Ethernet56,Ethernet60,Ethernet64,Ethernet68,Ethernet72,Ethernet76,Ethernet8"
```
Comment (ZhaohuiS) [inline tests/common/templates/default_acl_rules.json:27]: @yutongzhang-microsoft use the same interfaces for different topology?

Comment (yutongzhang-microsoft) [reply tests/common/templates/default_acl_rules.json:27]: No, this is set by `pre_acl_rules`

**Comment by yutongzhang-microsoft** [comment]
The failure of pipeline is caused by python3 migration. 

    @pytest.fixture(scope="module")
    def enum_rand_one_per_hwsku_frontend_hostname(request):
>       return request.param
E       AttributeError: 'SubRequest' object has no attribute 'param'

File: tests/common/templates/default_acl_rules.json [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+                                "config": {
+                                    "sequence-id": 1
+                                },
+                                "l2": {
+                                    "config": {
+                                        "ethertype": "2048",
+                                        "vlan_id": "1000"
+                                    }
+                                },
+                                "input_interface": {
```
[the remark this answers is not in the document]
Comment (yutongzhang-microsoft) [reply tests/common/templates/default_acl_rules.json:23]: No, this template only be used by backend topo, and backend topo support this config.

**Review by ZhaohuiS — APPROVED** [review]
LGTM

**Pull request merged**